src/main/preferences/index.js

```diff
@@ -27,33 +39,37 @@ class Preference extends EventEmitter {
   constructor (paths) {
     super()
 
-    const { userDataPath, preferencesFilePath } = paths
+    const { userDataPath } = paths
     this._userDataPath = userDataPath
+    this.store = new Store({
+      schema
+    })
 
-    this.cache = null
-    this.staticPath = path.join(__static, 'preference.md')
-    this.settingsPath = preferencesFilePath
+    this.staticPath = path.join(__static, 'preference.json')
     this.init()
   }
 
-  init () {
-    const { settingsPath, staticPath } = this
-    const defaultSettings = this.loadJson(staticPath)
-    let userSetting = null
-
-    // Try to load settings or write default settings if file doesn't exists.
-    if (!fs.existsSync(settingsPath) || !this.loadJson(settingsPath)) {
-      ensureDirSync(this._userDataPath)
-      const content = fs.readFileSync(staticPath, 'utf-8')
-      fs.writeFileSync(settingsPath, content, 'utf-8')
-
-      userSetting = this.loadJson(settingsPath)
+  init = () => {
+    let defaultSettings = null
+    try {
+      defaultSettings = fse.readJsonSync(this.staticPath)
       if (isDarkSystemMode()) {
-        userSetting.theme = 'dark'
+        defaultSettings.theme = 'dark'
       }
-      this.validateSettings(userSetting)
+      defaultSettings = validate(defaultSettings)
+    } catch (err) {
+      log(err)
+    }
+
+    if (!defaultSettings) {
+      console.error('Can not load static preference.json file')
+      return
```

Comment:
Should we fail hard when we cannot get the default settings?

Reply:
Throw an exception to crash Mark Text. Otherwise Mark Text will crash later because no settings are set, right?